numpy/f2py/cfuncs.py

```diff
@@ -623,8 +623,9 @@
 cfuncs['try_pyarr_from_string'] = """\
 static int try_pyarr_from_string(PyObject *obj,const string str) {
     PyArrayObject *arr = NULL;
-    if (PyArray_Check(obj) && (!((arr = (PyArrayObject *)obj) == NULL)))
-        { STRINGCOPYN(PyArray_DATA(arr),str,PyArray_NBYTES(arr)); }
+    if (PyArray_Check(obj) && (!((arr = (PyArrayObject *)obj) == NULL))) {
+        STRINGCOPYN(PyArray_DATA(arr),str,PyArray_NBYTES(arr));
+    }
     return 1;
```

Comment:
Sorry, not directly related, but it looks like this return should be inside the if statement? (so that the error can be reached)